### DISCUSSION TITLE
Welcome to the Community Datasync Toolkit!

### DISCUSSION THREAD

**Discussion by adrianhall** [CommunityToolkit/Datasync#3] [Announcements]
The Community Datasync Toolkit is a new project under the dotnet Community Toolkit umbrella and a part of the dotnet Foundation.  It is based on the old "Azure Mobile Apps" code base, and can be considered the logical successor to it.

Feel free to ask questions here within discussion.

If you find a bug, please file an issue and include all the details for a reproduction of the issue by the maintainers.

If you want to contribute, please use a pull request.

We appreciate you being here, and look forward to collaborating!

**Comment by rgueldenpfennig** [comment]
Hey Adrian,
first of all thanks for your efforts and contributions regarding all things Datasync.
As the official Microsoft documentation is still mentioning that the Datasync library is the recommended way to establish offline data sync, what are your plans in regards of a roadmap?

I'm currently looking for a generic solution to implement offline data sync for a MAUI based app inlcuding Blazor Hybrid. But as one of the main requirements is support of Native AOT, I'm currently not able to choose the Datasync library. Do you have any plans for that? E.g. removing Newtonsoft.Json?

**Reply by adrianhall** [reply]
Great question.

First off, what's the CURRENT PLAN:
- .NET 8 first ASP.NET Core service that mirrors the original Azure Mobile Apps service.
- .NET 8 library that supports MAUI, WinUI3 explicitly, with follow ons for Uno and Avalonia if those teams step up to assist, using Entity Framework Core as the basis (i.e. offline first, rather than offline optional)
- .NET 8 library for Blazor Hybrid and Blazor WASM.

Right now, I'm working on the first two - explicitly the push/pull in the second one.  I'm also looking to fix some of the complaints about the original code such as "millisecond updatedAt causes incomplete syncs due to too many records" - mostly by switching over to nanosecond and using a long as a transmission mechanism,

What's the FUTURE PLAN?

Well, that's up to you.  I'm obviously going to be doing the library for me first and everyone else next.  However, I'm happy to work on  issues as they come up, extensions that are highly voted, etc.  So, no plans right now.  However, I've got some ideas that I may put down on paper.

**Reply by dave-murray** [reply]
Hi Adrian,

Are you using MAUI here as shorthand that includes .NET for Android & .NET for iOS? We have a large Android application that we are currently converting from Xamarin Android to .NET 8 for Android. Our next plan is to replace our current home grown data sync framework and we have been evaluating Azure Mobile Apps. I'd like some reassurance that the Community Toolkit version will work with .NET 8 for Android before doing any real work on this project since we've already had problems replacing a number of Xamarin Android libraries that have no .NET for Android equivalent.

**Reply by adrianhall** [reply]
There is nothing explicitly in the library for MAUI, so it should "just work".  That being said, I do not have a .NET for Android app and it's not an environment I've worked in.  I would appreciate a sample (of both iOS and Android) to work with - maybe a conversion of the Todo app?

**Reply by dave-murray** [reply]
@adrianhall I have some time this week so I'll try porting the Xamarin Android ToDo sample to .NET for Android. Unfortunately I don't have an iOS build toolchain set up atm but I may be able to look into that later.

**Reply by adrianhall** [reply]
Thanks @dave-murray ! The work is much appreciated!
